playground/gltf-Pointer.ts

```diff
@@ -0,0 +1,241 @@
+  Entity,
+  GLTFResource,
+  Matrix,
+  MeshRenderer,
+  PBRMaterial,
+  PrimitiveMesh,
+  Renderer,
+  Script,
+  SkyBoxMaterial,
+  StaticCollider,
+  UnlitMaterial,
+  Vector3,
+  WebGLEngine
+} from "oasis-engine";
+import { LitePhysics } from "@oasis-engine/physics-lite";
+import * as dat from "dat.gui";
+
+enum MergeModeType {
+  merge,
+  dontMerge
+}
+class glTFCollider extends Script {
+  private _curClickRenderer: Renderer;
+  private _cycleTime: number = 2500;
+  private _shapeSize: Vector3 = new Vector3();
+  private _shapePosition: Vector3 = new Vector3();
+  private _auxiliaryBox: Renderer[] = [];
+
+  private _mergeMode: MergeModeType = MergeModeType.dontMerge;
+  private _mergeEntity: Entity;
+  private _notMergeEntity: Entity[] = [];
+
+  private _tempColor: Color = new Color();
+  private _tempMatrix: Matrix = new Matrix();
+
+  set mergeMode(type: MergeModeType) {
+    if (this._mergeMode !== type) {
+      this._mergeMode = type;
+      const { _notMergeEntity: notMergeEntity, _mergeEntity: mergeEntity } = this;
+      const entityLen = notMergeEntity.length;
+      this._resetClickRenderer(null);
+      switch (type) {
+        case MergeModeType.merge:
+          for (let i = entityLen - 1; i >= 0; i--) {
+            notMergeEntity[i].isActive = false;
+          }
+          if (mergeEntity) {
+            mergeEntity.isActive = true;
+          } else {
+            this.mergeBoundBox();
+          }
+          break;
+        case MergeModeType.dontMerge:
+          mergeEntity.isActive = false;
+          if (entityLen > 0) {
+            for (let i = entityLen - 1; i >= 0; i--) {
+              notMergeEntity[i].isActive = true;
+            }
+          } else {
+            this.dontMergeBoundBox();
+          }
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  mergeBoundBox() {
+    const renderers = this.entity.getComponentsIncludeChildren(MeshRenderer, []);
+    const boundingBox = renderers[0].bounds.clone();
+    for (let i = renderers.length - 1; i > 0; i--) {
+      BoundingBox.merge(boundingBox, renderers[i].bounds, boundingBox);
+    }
+    this._mergeEntity = this._addBoundingBox(boundingBox);
+  }
+
+  dontMergeBoundBox() {
+    const { _notMergeEntity: notMergeEntity } = this;
+    if (notMergeEntity.length === 0) {
+      const renderers = this.entity.getComponentsIncludeChildren(MeshRenderer, []);
+      for (let i = renderers.length - 1; i >= 0; i--) {
+        notMergeEntity.push(this._addBoundingBox(renderers[i].bounds, renderers[i]));
+      }
+    }
+  }
+
+  onStart(): void {
+    this.dontMergeBoundBox();
+  }
+
+  onUpdate(): void {
+    if (this._curClickRenderer) {
+      const v = (Math.sin((2 * Math.PI * (engine.time.nowTime % this._cycleTime)) / this._cycleTime) + 1) / 2;
+      this._setRendererColor(this._curClickRenderer, this._tempColor.setValue(v, 0, 0, 1));
+    }
+  }
+
+  private _addBoundingBox(boundingBox: BoundingBox, renderer?: Renderer): Entity {
```

Comment:
These codes are too hard to read for newbies. We should write another simple demo to show how to add input event to glTF entity.